modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -8,33 +8,31 @@ class MviUsersController < ApplicationController
       skip_before_action :authenticate
       before_action :check_required_headers
 
+      # Allows MVI lookups based on user identity traits (presumably obtained from the upstream
+      # identity provider). Returns a 200 response when the MVI lookup succeeds. Raises an error
+      # otherwise.
       def show
-        user = build_user(build_identity)
+        user_identity = save_requested_identity
```

Comment:
it seems a bit strange to me that a `show` action saves data  from the request to redis.

Reply:
i agree, show action should not be doing a create or update.

Reply:
I agree that it seems strange but all of the other arrangements I can envision also seem strange. E.g. Using a `GET` method to reach a `create` action or using a `POST` method with an empty body. Removing the save call from this action is conceptually an option. However, since the `OpenidApplicationController` saves a user model whenever it establishes a session for an openid token, we will very likely see a plethora of `show` and `index` calls saving an `OpenidUser`/`OpenidUserIdentity` model during the auth phase.

Reply:
This is just a thought, but I think if this were abstracted down into a class it might be more clear what the intention is. Or maybe you could just help me to better understand this. 

It seems like you are fethching certain attributes from the request header and then using those attributes to build an object. Shouldn't that live as part of the authentication method or callbacks elsewhere? Why a separate controller?